pybind_interface/pybind_main.cpp

```diff
@@ -489,6 +556,132 @@ py::array_t<float> qsim_simulate_fullstate(
   return py::array_t<float>(fsv_size, fsv, capsule);
 }
 
+py::array_t<float> qtrajectory_simulate_fullstate(const py::dict &options,
+                                                  uint64_t input_state) {
+  NoisyCircuit<Cirq::GateCirq<float>> ncircuit;
+  unsigned num_qubits;
+  try {
+    ncircuit = getNoisyCircuit(options);
+    num_qubits = qsim::count_qubits(ncircuit);
+  } catch (const std::invalid_argument &exp) {
+    IO::errorf(exp.what());
+    return {};
+  }
+
+  using Simulator = qsim::Simulator<For>;
+  using StateSpace = Simulator::StateSpace;
+  using State = StateSpace::State;
+  using Runner = qsim::QuantumTrajectorySimulator<IO, Cirq::GateCirq<float>,
+                                                  MultiQubitGateFuser,
+                                                  Simulator>;
+
+  Runner::Parameter param;
+  uint64_t seed;
+  try {
+    param.num_threads = parseOptions<unsigned>(options, "t\0");
+    param.max_fused_size = parseOptions<unsigned>(options, "f\0");
+    param.verbosity = parseOptions<unsigned>(options, "v\0");
+    seed = parseOptions<unsigned>(options, "s\0");
+  } catch (const std::invalid_argument &exp) {
+    IO::errorf(exp.what());
+    return {};
+  }
+
+  StateSpace state_space(param.num_threads);
+
+  float *fsv;
+  const uint64_t fsv_size = std::pow(2, num_qubits + 1);
+  const uint64_t buff_size = state_space.MinSize(num_qubits);
+  if (posix_memalign((void **)&fsv, 32, buff_size * sizeof(float))) {
```

Comment:
This needs posix platforms and might not work on things like windows. See https://github.com/quantumlib/qsim/blob/master/lib/vectorspace.h#L74

Reply:
Opened #289 to track this. If we get a fix for that before this goes in, I'll update this PR to match - otherwise leaving as-is.

Reply:
#290 can be used to fix it. Replace
```C++
  float *fsv;
  const uint64_t fsv_size = std::pow(2, num_qubits + 1);
  const uint64_t buff_size = state_space.MinSize(num_qubits);
  if (posix_memalign((void **)&fsv, 32, buff_size * sizeof(float))) {
    IO::errorf("Memory allocation failed.\n");
    return {};
  }

  State state = state_space.Create(fsv, num_qubits);
```
with
```C++
  State state = state_space.Create(num_qubits);
```
and add `float* fsv = state.release();` after `state_space.InternalToNormalOrder(state);`. One thing is that `delete` is used to deallocate this memory. It seems it works with `posix_memalign`. I don't know if it works on Windows. Perhaps `delete reinterpret_cast<float *>(data);` can be replaced with `detail::free(data);`.

Reply:
Done.